internal/engine/eval/vulncheck/vulncheck.go

```diff
@@ -60,9 +84,16 @@ func (e *Evaluator) Eval(ctx context.Context, pol map[string]any, res *engif.Res
 	}
 
 	if len(vulnerablePackages) > 0 {
-		return evalerrors.NewDetailedErrEvaluationFailed(
-			templates.VulncheckTemplate,
-			map[string]any{"packages": vulnerablePackages},
+		if e.featureFlags != nil && flags.Bool(ctx, e.featureFlags, flags.ImprovedEvalDetails) {
```

Comment:
Should we improve `flags.Bool` to handle a `nil` IClient?  It sort of feels better than littering `nil` checks elsewhere in the code.

Reply:
I'm okay with that, but I'd also be okay with `nil` Client == `false` result, always.

Needing to think about default polarity of flags in each case is probably an anti-pattern -- I'd rather (like Go) use a default-false polarity, and flags express themselves as positive assertions of some variation from the base behavior.

Reply:
I don't think reasoning about default-false polarity when writing code is the problem, I think the problems begin when trying to figure out why some user facing element, far away from the flag evaluation, does not have the expected form because of Go-style default value semantics.

In general, I'm not a big fan of default values as a solution to memory initialization either.